modules/libcom/src/as/asLib.h

```diff
@@ -42,21 +42,28 @@ typedef void (*ASCLIENTCALLBACK) (ASCLIENTPVT,asClientStatus);
 /* The following  routines are macros with the following syntax
 long asCheckGet(ASCLIENTPVT asClientPvt);
 long asCheckPut(ASCLIENTPVT asClientPvt);
+long asCheckRPC(ASCLIENTPVT asClientPvt);
 */
 #define asCheckGet(asClientPvt) \
     (!asActive || ((asClientPvt)->access >= asREAD))
 #define asCheckPut(asClientPvt) \
     (!asActive || ((asClientPvt)->access >= asWRITE))
+#define asCheckRPC(asClientPvt) \
+    (!asActive || ((asClientPvt)->access >= asRPC))
 
 /* More convenience macros
 void *asTrapWriteWithData(ASCLIENTPVT asClientPvt,
-     const char *userid, const char *hostid, void *addr,
+     const char *userid, char *method, char *authority, const char *hostid, void *addr,
      int dbrType, int no_elements, void *data);
 void asTrapWriteAfter(ASCLIENTPVT asClientPvt);
 */
-#define asTrapWriteWithData(asClientPvt, user, host, addr, type, count, data) \
+#define asTrapWriteWithData(asClientPvt, user, host, isTLS, addr, type, count, data) \
     ((asActive && (asClientPvt)->trapMask) \
-    ? asTrapWriteBeforeWithData((user), (host), (addr), (type), (count), (data)) \
+    ? asTrapWriteBeforeWithDataX((user), "ca", NULL, (host), (isTLS), (addr), (type), (count), (data)) \
+    : 0)
+#define asTrapWriteWithDataX(asClientPvt, user, method, authority, host, isTLS, addr, type, count, data) \
+    ((asActive && (asClientPvt)->trapMask) \
+    ? asTrapWriteBeforeWithDataX((user), (method), (authority), (host), (isTLS), (addr), (type), (count), (data)) \
```

Comment:
Firstly, let's add a `static inline` function instead of another "convenience" macro.  Secondly, maybe switch from passing an increasingly long argument list to a `struct`?